File: javav2/example_code/sts/src/main/java/com/example/sts/GetSessionToken.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.sts;

// snippet-start:[sts.java2.get_session_token.main]
// snippet-start:[sts.java2.get_session_token.import]
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.sts.StsClient;
import software.amazon.awssdk.services.sts.model.GetSessionTokenResponse;
import software.amazon.awssdk.services.sts.model.StsException;
import software.amazon.awssdk.services.sts.model.GetSessionTokenRequest;
// snippet-end:[sts.java2.get_session_token.import]

/**
 * Before running this Java V2 code example, set up your development
 * environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class GetSessionToken {
    public static void main(String[] args) {
        Region region = Region.US_EAST_1;
        StsClient stsClient = StsClient.builder()
                .region(region)
                .build();

        getToken(stsClient);
        stsClient.close();
    }

    public static void getToken(StsClient stsClient) {
        try {
            GetSessionTokenRequest tokenRequest = GetSessionTokenRequest.builder()
                    .durationSeconds(1500)
                    .build();

            GetSessionTokenResponse tokenResponse = stsClient.getSessionToken(tokenRequest);
            System.out.println("The token value is " + tokenResponse.credentials().sessionToken());

        } catch (StsException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}
// snippet-end:[sts.java2.get_session_token.main]

File: kotlin/services/sts/src/main/kotlin/com/kotlin/sts/GetSessionToken.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.kotlin.sts

// snippet-start:[sts.kotlin.get_session_token.import]
import aws.sdk.kotlin.services.sts.StsClient
import aws.sdk.kotlin.services.sts.model.GetSessionTokenRequest
// snippet-end:[sts.kotlin.get_session_token.import]

/**
Before running this Kotlin code example, set up your development environment,
including your credentials.

For more information, see the following documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */
suspend fun main() {
    getToken()
}

// snippet-start:[sts.kotlin.get_session_token.main]
suspend fun getToken() {
    val request =
        GetSessionTokenRequest {
            durationSeconds = 1500
        }

    StsClient { region = "us-east-1" }.use { stsClient ->
        val tokenResponse = stsClient.getSessionToken(request)
        println("The token value is ${tokenResponse.credentials?.sessionToken}")
    }
}
// snippet-end:[sts.kotlin.get_session_token.main]
